### PR TITLE
Add `containerd_namespaces` parameter

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Introduce `containerd_namespaces` param, alias for `containerd_namespace`.

### Motivation

Follow-up on https://github.com/DataDog/datadog-agent/pull/10509 where `containerd_namespace` became a list, now `containerd_namespaces` would make more sense.

### Describe how to test/QA your changes

Same as https://github.com/DataDog/datadog-agent/pull/10509 and https://github.com/DataDog/datadog-agent/pull/10128 but with `DD_CONTAINERD_NAMESPACES` instead of `DD_CONTAINERD_NAMESPACE`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -2446,6 +2446,16 @@ api_key:
 # containerd_namespace:
 #   - k8s.io
 
+## @param containerd_namespaces - list of strings - optional - default: []
+## @env DD_CONTAINERD_NAMESPACE - space separated list of strings - optional - default: []
```
Comment (davidor) [inline pkg/config/config_template.yaml:2450]: `DD_CONTAINERD_NAMESPACES`

**Review by davidor — APPROVED** [review]

**Comment by vboulineau** [comment]
I believe properly setting `containerd_namespaces` though an [override function](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_overrides.go#L11) would be cleaner as it abstracts away the migration from the `containerd` package and ensures anyone will always get the proper value anywhere.

I also believe the `containerd_namespace` variable should be removed from the `config_template.yaml` file to avoid confusions.

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -2446,6 +2446,16 @@ api_key:
 # containerd_namespace:
 #   - k8s.io
 
+## @param containerd_namespaces - list of strings - optional - default: []
+## @env DD_CONTAINERD_NAMESPACES - space separated list of strings - optional - default: []
+## Activating the Containerd check also activates the CRI check, as it contains an additional subset of useful metrics.
+## Defaults to [] which configures the agent to report metrics and events from all the containerd namespaces.
+## containerd_namespaces acts as an alias for containerd_namespace. When both containerd_namespaces and containerd_namespace
+## are configured, the Agents merges the two lists.
```
Comment (maycmlee) [inline pkg/config/config_template.yaml:2455]:
```suggestion
## are configured, the Agent merges the two lists.
```

**Review by maycmlee — APPROVED** [review]
Small fix

**Comment by ahmed-mez** [comment]
> I believe properly setting containerd_namespaces though an [override function](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_overrides.go#L11) would be cleaner as it abstracts away the migration from the containerd package and ensures anyone will always get the proper value anywhere.

Thanks for the suggestion! wasn't aware of the override logic in the config pkg. I made the change!

> I also believe the containerd_namespace variable should be removed from the config_template.yaml file to avoid confusions.

I added `## Deprecated - use containerd_namespaces instead` on top of that section 👍  (Inspired by what we do already for another field)

File: pkg/config/config.go [hunk 3]
```diff
@@ -570,6 +570,7 @@ func InitConfig(config Config) {
 
 	// Containerd
 	config.BindEnvAndSetDefault("containerd_namespace", "")
+	config.BindEnvAndSetDefault("containerd_namespaces", []string{}) // alias for containerd_namespace
```
Comment (djmitche) [inline pkg/config/config.go:573]: If the singular is deprecated, isn't the aliasing going the other direction?

**Review by djmitche — APPROVED** [review]

**Pull request merged**